k8s/deployments.go

```diff
@@ -90,6 +95,61 @@ func (c deployments) CreateWelcomeDeployment(app *models.App) error {
 	return err
 }
 
+func (c deployments) create(
+	w io.WriteCloser,
+	app *models.App,
+	file *runtime.File,
+	deploy *deploy,
+	storage helpers.Storage,
+	done chan<- struct{},
+) {
+	// create log context with uuid and app name
+	lc := log.WithField("app", app.Name).WithField("deployUUID", deploy.uuid)
+	lc.Info("starting deploy...")
+	defer close(done)
+
+	// parsing teresa.yaml and updating app
+	if err := parsingTeresaYaml(app, file.Data); err != nil {
```

Comment:
Maybe you have just fixed a strange periodic S3 related bug with this last suggestion :+1: